File: eAnimalShelter/UI/eanimalshelter_mobile/lib/models/user_update_request.dart
class ProfileUpdateRequest {
  String firstName;
  String lastName;
  String email;
  String? phoneNumber;
  String? address;

  ProfileUpdateRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    this.phoneNumber,
    this.address,
  });

  Map<String, dynamic> toJson() {
    return {
      "firstName": firstName,
      "lastName": lastName,
      "email": email,
      "phoneNumber": phoneNumber,
      "address": address,
    };
  }
}

File: eAnimalShelter/eAnimalShelter.Model/Requests/UserUpdateRequest.cs
namespace eAnimalShelter.Model.Requests
{
    public class UserUpdateRequest
    {
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string Email { get; set; } = null!;

        public string? PhoneNumber { get; set; }
        public string? Address { get; set; }

        public bool IsActive { get; set; }
        public int RoleId { get; set; }
    }
}